camayoc/tests/qpc/api/v1/utils.py

```diff
@@ -157,11 +161,13 @@ def wait_until_state(scanjob, timeout=120, state='completed'):
 
     while (not scanjob.status() or not scanjob.status()
             == state) and timeout > 0:
-        if state == 'stopped' and scanjob.status() in QPC_SCAN_TERMINAL_STATES:
+        if state in stopped_states and scanjob.status() in stopped_states:
             # scanjob is no longer running, so we will return
             return
         time.sleep(5)
         timeout -= 5
+        if scanjob.status() == state:
```

Comment:
This condition is already on the while loop condition, I think we should either check inside the loop of on the loop condition.